src/Symfony/Component/Validator/Tests/ConstraintValidatorTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+final class ConstraintValidatorTest extends TestCase
+{
+    /**
+     * @dataProvider formatValueProvider
+     */
+    public function testFormatValue($expected, $value, $format = 0)
+    {
+        $this->assertSame($expected, (new TestFormatValueConstraintValidator())($value, $format));
+    }
+
+    public function formatValueProvider(): array
+    {
+        $data = [
+            ['true', true],
+            ['false', false],
+            ['null', null],
+            ['resource', fopen('php://memory', 'r')],
+            ['"foo"', 'foo'],
+            ['array', []],
+            ['object', $toString = new TestToStringObject()],
+            ['ccc', $toString, ConstraintValidator::OBJECT_TO_STRING],
```

Comment:
Is it correct, that this test depends on the test before?

Reply:
For me, it doesn't really depends of it. It just reuses the same object instance to avoid a new instantiation. It also highlights that the same object is formatted differently depending of the format parameter.